source/base/ReferenceCounter.ooc

```diff
@@ -1,20 +1,30 @@
 import threading/Thread
 import Synchronized
 
-ReferenceCounter: class extends Synchronized {
+ReferenceCounter: class {
 	_target: Object
 	count: static Int = 0
 	_count: Int = 0
 	_kill := false
-	init: func (target: Object) {
-		super()
+	_lock: Mutex
+	isSafe: Bool {
+		get { !this _lock instanceOf?(MutexUnsafe) }
+		set(value) {
+			if (this isSafe != value) {
+				this _lock free()
```

Comment:
I just realized that... now that this class can be initialized with a `MutexGlobal`, if anyone actually *does* initialize one with a `MutexGlobal`, and then calls `isSafe = true` *or* `isSafe =false`, this line might cause problems... I'm not sure what happens if you try to call `free` on a `MutexGlobal`.